src/pdfviewer/PDFDocument.cpp

```diff
@@ -4298,11 +4298,15 @@ PDFDocument *PDFDocument::findDocument(const QString &fileName)
 
 void PDFDocument::saveGeometryToConfig()
 {
-	globalConfig->windowLeft = x();
```

Comment:
if the window position is only saved when not maximised, then the move(x,y) is possibly undefined or wrong.
This probably comes into play for multi-monitor set-ups ...

Reply:
What I've seen with debug under Win10 lead me to the conclusion that retrieving window coordinates while in maximized (and similar) modes returns invalid values. Especially the coordinate of the upper left window corner has negative values. It seems that line 2860 was introduced to fix this. So I assume that maximized/fullscreen needs to be handled by flags not by coordinates.

Be aware that the code retrieves coordinates from different things: The window corner as mentioned before and the size of the pdf widget. This can introduce problems because we and Qt have no access to the geometry of the window decoration (how thick are the border lines of the window surrounding the widget, height of the window title, and so on, which are under control of the window manager). Typically we don't need to know these details when all changes to the coordinates (and flags) are retrieved. Because with these values we should be able to restore a correct window at any time. So the move should be no problem here. Can you reproduce some case?

But this is currently not the case. After my PR I observed following: Maximize and restore pdf window. Thus window coordinates are retrieved and stored. Now move or resize the window, then maximize it, then end txs. When starting txs again the maximized window mode is restored with my PR, but when the maximized flag is cleared, then the normal window has the last saved coordinates instead of the last ones used. I tried to improve this with following code where I need help:
```
connect(window(), SIGNAL(PDFDocument::windowStateChanged(Qt::WindowState)), this, SLOT(slotWindowStateChanged(Qt::WindowState)));
```
But I couldn't figure out how to find the window as the sender of the signal. If this would work then the current coordinates of the window could be retrieved by txs on maximizing the window (and the like). This should be sufficient for single monitor setups.

In multi monitor setups this may lead to cases where the window is placed on the wrong screens. And what about the case that one drags the maximized window to another screen such that it appears maximized (or normal) there?